apps/ledger-live-mobile/src/logic/postOnboarding/useNavigateToPostOnboardingHubCallback.ts

```diff
@@ -1,12 +1,44 @@
 import { useNavigation } from "@react-navigation/native";
 import { useCallback } from "react";
+import { RootNavigation } from "../../components/RootNavigator/types/helpers";
 import { NavigatorName, ScreenName } from "../../const";
 
 export function useNavigateToPostOnboardingHubCallback() {
   const navigation = useNavigation();
-  return useCallback(() => {
-    navigation.navigate(NavigatorName.PostOnboarding, {
-      screen: ScreenName.PostOnboardingHub,
-    });
-  }, [navigation]);
+  return useCallback(
+    (resetNavigationStack?: boolean) => {
+      if (resetNavigationStack) {
+        (navigation as unknown as RootNavigation).reset({
+          index: 0,
+          routes: [
+            {
+              name: NavigatorName.Base,
+              state: {
+                routes: [
+                  {
+                    name: NavigatorName.Main,
```

Comment:
when closing the PostOnboarding navigator it will go to the Main navigator

Reply:
This is horrible